src/forms/CurrentExpenses.js

```diff
@@ -41,20 +40,36 @@ import { getUnder13Expenses } from '../utils/cashflow';
 // ========================================
 // COMPONENTS
 // ========================================
-const EarnedFrom = function ({ hasExpenses, cashflowProps, children }) {
+const EarnedFrom = function ({ hasExpenses, CashFlowRow, label, propData }) {
+
+  /** @todo Save amount temporarily when 'source'
+   *      amount is set to 0. */
+  var reset = function (evnt) {
+    var { name, update } = propData;
+
+    update(evnt, {
+      name:  name,
+      value: 0,
+    });
+  };
 
   if (hasExpenses) {
 
-    // Because we're familiar with the benefit code, we
-    // happen to know these values don't need to be reset
-    // to 0, even if the client erases childcare expenses.
-    // Not sure if that's a great general practice, though.
+    var { name, client } = propData;
+    var showProps = {
+      propName: name,
+      show:     client[ name ] > 0,
```

Comment:
I think there's a basic misunderstanding here. Rendering `ShowOnYes`, when first triggered, shows only the 'yes/no' radio buttons, not the cash flow inputs. For example, if the client enters any child care expenses, the 'yes/no' radio inputs will be shown to ask if their income is affected by that child care. If they then select 'yes', the children given to `ShowOnYes` will be rendered (in this case cash flow inputs) for more interaction.

The `show` prop handed into the `ShowOnYes` component is for when a user goes to a section (e.g. the 'Expenses' section), enters some data, like child care expenses (revealing `ShowOnYes`), then enters, for example, some income amount that's generated because of those expenses. They then navigate away from that section and navigate back. They still have those child care expenses, so `ShowOnYes` should still be visible, but they also still have that affected income amount, so the cash flow inputs should also be initially visible.

It's now been renamed to `showOnStart`, though maybe that's not going to clarify things enough. I'll try to come up with another name, I guess. `showChildrenOnStart`?

Another option would be to store something like `confirm_ShowEarnedFromChildCare` to keep track of whether they answered 'yes' or 'no' and use that to decide whether to show the children. If it's at all possible to make it clear what's going on, though, I'd rather use some preexisting prop so we don't clutter `client` any more than is needed. I'll work on clarifying stuff some more.

Thanks to both of you, @wacii and @ethanbb , for hitting me over the head with this until I understand that the problem is with the clarity of the code and needs a code fix rather than an explanation.